src/handlers/initState.js

```diff
@@ -41,6 +45,9 @@ export default async function initState(params) {
           edge.node.text.replace(/\s/g, ''),
           event.input.replace(/\s/g, '')
         );
+        // Track which Article is searched.
+        ga(userId, { ec: 'Article', ea: 'Search', el: edge.node.id });
+
```

Comment:
`.map` calls are not supposed to contain side effects (such as calling other APIs).

Let's do impression tracking in a dedicated for-each loop. We can put this along side with the `ArticleFound` event above, so that similar logics are located in the same place.

Also, since we are sending one Article-Search event for each found article, let's make it [non-interactive](https://support.google.com/analytics/answer/1033068#NonInteractionEvents) so that we can get better bounce rate calculations. We can achieve this by using [`ni`](https://github.com/peaksandpies/universal-analytics/blob/master/AcceptableParams.md#non-interaction-hit) param.